### PR TITLE
Integrate browser-toolbar component.

### PR DESCRIPTION
**Pull request by pocmo**
This is a work-in-progress version of a `browser-toolbar` integration. It's not feature complete yet, but it does allow using the browser (editing & displaying).

### CODE REVIEW

**Comment by pocmo** [comment]
@jonalmeida Okay, I think this is ready to review :) For the TP icon to fully work, we need #4960 and the latest A-C Nightly.

File: app/src/main/java/org/mozilla/focus/browser/binding/BlockingThemeBinding.kt [hunk 1]
```diff
@@ -29,7 +30,7 @@ class BlockingThemeBinding(
     private val isCustomTab: Boolean,
     private val statusBar: View,
     private val urlBar: View,
-    private val blockView: View
+    private val toolbar: BrowserToolbar
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/browser/binding/BlockingThemeBinding.kt:33]: unused?

File: app/src/main/java/org/mozilla/focus/browser/integration/NavigationButtonsIntegration.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
+package org.mozilla.focus.browser.integration
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/browser/integration/NavigationButtonsIntegration.kt:1]: nit: license

File: app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                store,
+                toolbar,
+                sessionId = customTabId,
+                useCases = customTabsUseCases,
+                closeListener = { fragment.closeCustomTab() }
+            )
+        }
+
+        if (HardwareUtils.isTablet(context)) {
+            navigationButtonsIntegration = NavigationButtonsIntegration(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt:67]: Nice!

File: app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsViewModel.kt [hunk 4]
```diff
@@ -107,6 +106,8 @@ class SearchSuggestionsViewModel(application: Application) : AndroidViewModel(ap
             State.Disabled(givePrompt)
         }
 
+        Log.w("SKDBG", " * State(${hashCode()}) = $state")
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsViewModel.kt:109]: Probably meant to be removed? :)

**Review by jonalmeida — APPROVED** [review]
This looks great!

Two things I observed:
 - The progress bar doesn't show up when the toolbar is hidden, but that should be easy to do in a follow-up.
 - We may break password managers that use accessibility services to read the url from the toolbar based on the ID of the view.

File: app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsViewModel.kt (hunk 4 above)
Comment (pocmo) [reply app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsViewModel.kt:109]: 🤣

**Comment by pocmo** [comment]
> * The progress bar doesn't show up when the toolbar is hidden, but that should be easy to do in a follow-up.

Yeah, I looked at Fenix and over there we keep the toolbar visible during loads (and in other situations), we should probably do the same in Focus.

> * We may break password managers that use accessibility services to read the url from the toolbar based on the ID of the view.

Possible, but I think there's no way around that.

**Pull request merged**